persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageInsertExtension.kt

```diff
@@ -204,6 +210,58 @@ internal class MessageInsertExtensionImpl(
         }
     }
 
+    private fun insertUnreadEvent(message: MessageEntity) {
+        if (!message.isSelfMessage) {
+            when (message.content) {
+                is MessageEntityContent.Knock -> unreadEventsQueries.insertEvent(
+                    message.id,
+                    UnreadEventTypeEntity.KNOCK,
+                    message.conversationId,
+                    message.date
+                )
+                is MessageEntityContent.Text -> {
+                    val textContent: MessageEntityContent.Text = message.content as (MessageEntityContent.Text)
+                    when {
+                        textContent.quotedMessage?.isQuotingSelfUser ?: false -> unreadEventsQueries.insertEvent(
+                            message.id,
+                            UnreadEventTypeEntity.REPLY,
+                            message.conversationId,
+                            message.date
+                        )
+                        textContent.mentions.map { it.userId }.contains(selfUserIDEntity) ->
+                            unreadEventsQueries.insertEvent(
+                                message.id,
+                                UnreadEventTypeEntity.MENTION,
+                                message.conversationId,
+                                message.date
+                            )
+                        else -> unreadEventsQueries.insertEvent(
+                            message.id,
+                            UnreadEventTypeEntity.MESSAGE,
+                            message.conversationId,
+                            message.date
+                        )
+                    }
```

Comment:
It would be a bit more readable if these blocks inside the `when` branches were extracted to its own function.

Reply:
done